src/kademlia/dht_storage.cpp

```diff
@@ -346,6 +345,7 @@ namespace
 		}
 
 		void put_immutable_item(sha1_hash const& target
+			, sha1_hash const& our_id
```

Comment:
I don't think it makes sense to partition the item storage by the id of the node which stored it. I would suggest having `dht_default_storage_constructor` take a reference to a distance function (probably as a `std::function`) rather than a reference to a node id. The distance function would compute the minimum distance from a given key to any of the client's node ids. The implementation should probably be as a member function of `dht_tracker`.